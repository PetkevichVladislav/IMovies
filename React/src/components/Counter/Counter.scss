.counter{
    margin: 0px;

    &__text{
        color: #FFFFFF;
        font-size: 20px;
    }
    
    &__count{
        display: inline;
        font-weight: 700;
    }

    &__message{
        display: inline;
        font-weight: 500;
        margin-left: 4px;
    }
}