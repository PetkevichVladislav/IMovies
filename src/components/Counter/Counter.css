.counter{
    margin: 0px;
}

.counter__text{
    color: #FFFFFF;
    font-size: 20px;
}

.counter__count{
    display: inline;
    font-weight: 700;
}

.counter__message{
    display: inline;
    font-weight: 500;
    margin-left: 4px;
}