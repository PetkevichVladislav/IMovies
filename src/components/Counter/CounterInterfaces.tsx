export interface ICounterProps {
    initialValue: number;
}

export interface ICounterState {
    count: number;
}