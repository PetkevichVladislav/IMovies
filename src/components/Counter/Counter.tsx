import "./Counter.css"

interface CounterProps {
    count: number,
}

function Counter({ count } : CounterProps)
{
    return (
        <div className="counter">
            <p className="counter__text counter__count">{count}</p>
            <p className="counter__text counter__message">movies found</p>
        </div>
    );
}

export default Counter;