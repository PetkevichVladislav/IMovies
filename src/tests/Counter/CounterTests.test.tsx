import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import Counter from '../../components/Counter/Counter';

describe("Counter", () => {
  it("When data is valid component should renders correctly", () => {
    const tree = renderer.create(<Counter count={0}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("When counter creates initial value should setups correctly", () => {
    render(<Counter count={5}/>);
    const expectedValue = screen.getByText(5);
    expect(expectedValue).toBeInTheDocument();
  });
});