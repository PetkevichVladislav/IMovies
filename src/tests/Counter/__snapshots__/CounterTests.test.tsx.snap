// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter When data is valid component should renders correctly 1`] = `
<div
  className="counter"
>
  <p
    className="counter__text counter__count"
  >
    0
  </p>
  <p
    className="counter__text counter__message"
  >
    movies found
  </p>
  <button
    className="button-increment"
    onClick={[Function]}
  >
    increment
  </button>
  <button
    className="button-decrement"
    onClick={[Function]}
  >
    decrement
  </button>
</div>
`;
